### PR TITLE
Show Changes for Streamlit

### PR DESCRIPTION
**Pull request by naterush**
# Description

I made some changes:
1. Make it so if we don't have an args update - we can still get dataframe names from the passed string arguments. If we didn't do this, we would generate invalid scripts in Streamlit anytime you passed a string argument.
2. Move the import statements to the Python code. Again, not having these results in an invalid script. 
3. Makes the preprocess steps generate import statements as well, again for valid code reasons.
4. Made it so we don't send render count messages on streamlit, as in apps with multiple mitosheets this effectively bricks the entire application. 
5. Changed it so you don't need to pass a key for the spreadsheet to work properly - it will automatically use a good one. 

I merged them into dev b/c I wanted to test in deployed version!

# Testing

Please provide a list of the ways you can "access" or use the functionality. Please try and be exhaustive here, and make sure that you test everything you list.

- [ ] I have tested this on real data that is reasonable and large
- [ ] If I changed the interaction with JupyterLab, I tested that it does not break other programs (like VS Code), and tested that it works "multiple times" in the same notebook.

# Documentation

Note if any new documentation needs to addressed or reviewed.

### CODE REVIEW

**Comment by aarondr77** [comment]
Getting this error where its not generating unique keys for my spreadsheet components 

https://github.com/mito-ds/mito/assets/18709905/4969dbee-fbf6-43eb-b279-ed6191f17df1

**Comment by aarondr77** [comment]
We aren't handling broken analyses correctly.

See in this video, I have two mitosheets -- the first one imports a df and the second one imports the result of the first one

1. Add a column in the first Mito
2. Delete it in the second Mito
3. Delete it in the first Mito
4. Notice that the second Mito should have an error message, but instead it just clears the analysis. 

https://github.com/mito-ds/mito/assets/18709905/b97d7c15-fa8b-44e2-9273-e521cdcf057e

**Comment by aarondr77** [comment]
Bug where not all changes to the generated dataframes trigger the code to rerun? 

I have this app configuration: 

Screenshot 2023-07-27 at 12 47 40 PM

1. In Mito 1, add a column. See it appears in Mito 2
2. Set the column formula in Mito 1. See it appears in Mito 2
**3. Rename the column in Mito 1. See that it does not update in Mito 2!**
4. Add a new column in Mito 1. See it appears in Mito 2 and that the previous rename now appears too!
5. Set the column name in Mito 1. See it **does not** appear in Mito 2. 
6. Set the column formula in Mito 1. See that it does appear in Mito 2 along with the updated name. 

https://github.com/mito-ds/mito/assets/18709905/fcb76571-4e5e-4746-859e-a202471c1a14

File: mitosheet/src/jupyter/code.tsx [hunk 1]
```diff
@@ -22,17 +18,19 @@ export function getCodeString(
...
-        return `${importStatement}; register_analysis("${analysisName}");
-${finalCode}`
+        analysisRegisterCode = `register_analysis("${analysisName}");`
     } else {
-        return `${importStatement}; # Analysis Name:${analysisName};
-${finalCode}`
+        analysisRegisterCode = `# Analysis Name:${analysisName};`
     }
+
+    return finalCode.replace('import *', `import *; ${analysisRegisterCode}`);
```
[the remark this answers is not in the document]
Comment (naterush) [reply mitosheet/src/jupyter/code.tsx:33]: No we can still do this!

**Pull request closed without merging**